### PR TITLE
VIH-10432 Update multi day hearing backend redesign

### PR DESCRIPTION
**Pull request by oliver-scott**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/VIH-10432

### Change description ###
Changes the update multi-day hearing BFF to call the new bookings api endpoint (UpdateHearingsInGroup) instead of looping over hearings and updating participants and endpoints.
Some refactoring has been done to re-use existing update hearings logic.

### CODE REVIEW

File: AdminWebsite/AdminWebsite.IntegrationTests/packages.lock.json [hunk 1]
```diff
@@ -102,8 +102,8 @@
       },
       "BookingsApi.Client": {
         "type": "Transitive",
-        "resolved": "1.50.16",
-        "contentHash": "wtMHaYtTO5QcURzKjE2ynd3cIBoboMWPSFDkZ4X3iBO67JPtSs5B8784g5HnHB7Cb50GOiVAbOB3qJDQeq2eZw==",
+        "resolved": "1.51.4-pr-0815-0024",
```
Comment (oliver-scott) [inline AdminWebsite/AdminWebsite.IntegrationTests/packages.lock.json:105]: TODO update after bookings api merge

File: charts/vh-admin-web/values.dev.template.yaml [hunk 2]
```diff
@@ -9,4 +9,4 @@ java:
     AZUREAD__REDIRECTURI: https://${SERVICE_FQDN}/home
     DOM1__POSTLOGOUTREDIRECTURI: https://${SERVICE_FQDN}/logout
     DOM1__REDIRECTURI: https://${SERVICE_FQDN}/home
-  
\ No newline at end of file
+    VHSERVICES__BOOKINGSAPIURL: https://vh-bookings-api-pr-815.dev.platform.hmcts.net/
```
Comment (oliver-scott) [inline charts/vh-admin-web/values.dev.template.yaml:12]: TODO update before merge

File: AdminWebsite/AdminWebsite/Controllers/HearingsController.cs [hunk 3]
```diff
@@ -424,29 +425,88 @@ private async Task UpdateMultiDayHearing(EditMultiDayHearingRequest request, Gui
                 hearingsToUpdate.AddRange(futureHearings);
             }
 
-            foreach (var hearing in hearingsToUpdate)
+            if (_featureToggles.UseV2Api())
```
Comment (will-craig) [inline AdminWebsite/AdminWebsite/Controllers/HearingsController.cs:428]: Do you reckon we could break up UpdateMultiDayHearing() into a few more methods to reduce the complexity? Perhaps create one method for the V2 logic and one for the else block

Comment (oliver-scott) [reply AdminWebsite/AdminWebsite/Controllers/HearingsController.cs:428]: that's a good idea, updated

**Review by will-craig — APPROVED** [review]

**Review by komaldabhi1509 — APPROVED** [review]

**Pull request merged**